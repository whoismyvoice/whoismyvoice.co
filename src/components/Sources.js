import React from 'react';
import { Link } from 'react-router';

const Sources = React.createClass({
  render() {
    return  <div className="page-block bright-green">
      <p className="pageIntroduction">
      	Every senator listed on this website is a listed co-sponsor of Senate Bill 1881. Senate Bill 1881 is a "bill to prohibit the federal funding of the Planned Parenthood Federation of America". (1)<br /><br />
      	The Congressmen listed on this website are the listed members of the House Freedom Caucus, who issued a press release via Congressman Jim Jordan on September 10, 2015:
      	<br /><br />
      	"Given the appalling revelations surrounding Planned Parenthood, we cannot in good moral conscience vote to send taxpayer money to this organization while still fulfilling our duty to represent our constituents. We must therefore oppose any spendig measure that contains funding for Planned Parenthood."
      	<br /><br />
      	<span className="resources">
      		Additional resource data was pulled directly from the following sources:
			<br />
      		1) <Link to="http://jordan.house.gov/news/documentsingle.aspx?DocumentID=397949">Congressman Jim Jordan's Press Release</Link><br />
			2) <Link to="https://www.congress.gov/bill/114th-congress/senate-bill/1881">Senate Bill S.1881</Link><br />
			3) <Link to="http://sunlightfoundation.com/">Sunlight Foundation </Link>
		</span>
      </p>
    </div>;
  }
});

export default Sources;
