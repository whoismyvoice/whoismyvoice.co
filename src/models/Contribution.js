// @flow

import PropTypes from 'prop-types';

/**
 * Defines the type alias for a `Contribution` record.
 */
export type ContributionRecord = {
  amount: number,
  legislatorId: string,
  organization: string,
};

/**
 * Defines the `PropTypes` validation structure for a `Contribution` record.
 */
export const PropType = PropTypes.shape({
  amount: PropTypes.number.isRequired,
  legislatorId: PropTypes.string.isRequired,
  organization: PropTypes.string.isRequired,
});
