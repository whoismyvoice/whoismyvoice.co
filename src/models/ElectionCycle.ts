export interface ElectionCycle {
  year: string;
  label: string;
}

interface ResponseElectionCycle {
  ElectionCycle: string;
  label: string;
}

export function compareElectionCycles(
  c1: ResponseElectionCycle,
  c2: ResponseElectionCycle
) {
  if (c1.ElectionCycle > c2.ElectionCycle) {
    return 1;
  } else if (c1.ElectionCycle < c2.ElectionCycle) {
    return -1;
  } else {
    return 0;
  }
}
