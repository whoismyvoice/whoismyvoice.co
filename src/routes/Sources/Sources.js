import React, { Component } from 'react';

// Components
import FadedBG from '../../components/FadedBg';

class Sources extends Component {
  render() {
    return (
      <div className="page-block light-white">
        <FadedBG color="white" />
        <div className="page-wrapper">
          <h2>Sources</h2>
          <p className="page-introduction">
            The data on this website is based on the data released by the
            Federal Election Commission (FEC) on donations that were made during
            the 2015-2016 and 2017-2018 election cycles. Only donations
            exceeding $200 in value are required to be publicly disclosed, i.e.
            (multiple) donations below $200 threshold do not need to be
            itemized. WhoIsMyVoice.com uses APIs provided by{' '}
            <a href="https://maplight.org/">MapLight</a> to search FEC data.
            <br />
            <br />
            On December 3, 2015, the Senate voted down two amendments on gun
            control, just one day following the San Bernadino, California
            attacks. Senate Amendment (S.A) 2910 was a measure to provide
            responsible, consistent, universal background checks for gun
            purchases, and to ensure individuals listed in the National Instant
            Criminal Background Check System cannot buy a gun. This amendment
            included the same language as H.R 1076 which was voted down by
            Senate in April of 2013. Between April 2013 and December 2015, not
            one Senator changed their vote. Senate Amendment (S.A) 2908 was a
            measure to deny people on a federal terrorism watch list (a.k.a "the
            no-fly" list) the ability to purchase guns.
            <br />
            <br />
            Despite 12,472 deaths and 25,156 injuries from gun violence in 2015,
            12/3/15 was the first time any gun control measures whatsoever made
            it to the floor for a vote. In the case of the two amendments above,
            they only made it to vote because they were attached as amendments
            to the budget bill. (Data taken from{' '}
            <a
              href="http://www.gunviolencearchive.org/"
              target="_blank"
              rel="noopener noreferrer"
            >
              The Gun Violence Archive
            </a>).
          </p>
        </div>
      </div>
    );
  }
}

export default Sources;
