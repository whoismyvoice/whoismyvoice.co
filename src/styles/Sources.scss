@import '_variables';

.page-wrapper--sources ul.sources {
  list-style-type: square;
  text-align: left;
}

.page-wrapper--sources a {
  text-decoration: underline;

  &:visited {
    color: $dark-grey;
  }
  &:hover {
    color: $dark-orange-light;
  }
}
