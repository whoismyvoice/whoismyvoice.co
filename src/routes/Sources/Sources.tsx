import React from 'react';
import * as mixpanel from 'mixpanel-browser';

import { SOURCES_ROUTE } from '../../constants/mixpanel-events';
import FadedBG from '../../components/FadedBg';
import '../../styles/Sources.css';

class Sources extends React.Component {
  componentDidMount() {
    mixpanel.track.apply(mixpanel, SOURCES_ROUTE);
  }

  render() {
    return (
      <div className="page-block light-white">
        <FadedBG color="white" />
        <div className="page-wrapper page-wrapper--sources">
          <h2>Sources</h2>
          <p className="page-introduction">
            The data on this website is based on the data released by the
            Federal Election Commission (FEC) on donations that were made during
            the 2017-2018 and 2019-2020 election cycles. Only donations
            exceeding $200 in value are required to be publicly disclosed, i.e.
            (multiple) donations below $200 threshold do not need to be
            itemized. WhoIsMyVoice.com uses APIs provided by
            <a href="https://maplight.org/">MapLight</a> to search FEC data and
            the
            <a href="https://developers.google.com/civic-information">
              Google Civic Information API
            </a>
            to lookup representatives by address.
          </p>
          <h3>For More Information</h3>
          <ul className="sources">
            <li>
              <a href="https://www.cdc.gov/nchs/pressroom/sosmap/firearm_mortality/firearm.htm">
                CDC Statistics on Firearm Mortality By State
              </a>
            </li>
            <li>
              <a href="http://www.gunviolencearchive.org/">
                The Gun Violence Archive
              </a>
            </li>
            <li>
              <a href="https://everytownresearch.org/">
                Everytown for Gun Safety
              </a>
            </li>
            <li>
              <a href="https://home.nra.org/">National Rifle Association</a>
            </li>
          </ul>
        </div>
      </div>
    );
  }
}

export default Sources;
