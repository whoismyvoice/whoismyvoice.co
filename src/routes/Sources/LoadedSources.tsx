import React, { useEffect, useState } from 'react';

import { ELECTION_CYCLES } from '../../constants';
import { ElectionCycle } from '../../models/ElectionCycle';
import Sources from './Sources';

export function LoadedSources(): JSX.Element {
  const [cycles, setCycles] = useState<ElectionCycle[]>([]);
  useEffect(() => {
    ELECTION_CYCLES.then(cycles => setCycles([...cycles].reverse()));
  }, []);
  if (cycles.length === 0) {
    return <h1>Loading...</h1>;
  } else {
    return <Sources cycles={cycles} />;
  }
}
