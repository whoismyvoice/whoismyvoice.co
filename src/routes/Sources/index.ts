import { LoadedSources } from './LoadedSources';

export default LoadedSources;
