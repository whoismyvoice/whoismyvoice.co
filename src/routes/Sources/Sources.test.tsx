import React from 'react';
import { render } from '@testing-library/react';
import { ElectionCycle } from '../../models/ElectionCycle';
import { Sources } from './Sources';

const cycles: ElectionCycle[] = [
  { year: '2018', label: '2017-2018' },
  { year: '2020', label: '2019-2020' },
];

describe('Sources', () => {
  it('renders the page', () => {
    const { container, getByText } = render(<Sources cycles={cycles} />);
    const block = container.querySelector('.page-block');
    expect(block).not.toBeNull();
    expect(getByText('Sources').tagName).toMatch(/h2/i);
    expect(
      getByText('OpenSecrets and the Center for Responsive Politics').tagName
    ).toMatch(/a/i);
    expect(getByText('Google Civic Information API').tagName).toMatch(/a/i);
    expect(getByText('the @unitedstates project').tagName).toMatch(/a/i);
    expect(
      getByText('political spending reaches record numbers').tagName
    ).toMatch(/a/i);
    expect(getByText('For More Information on Gun Violence').tagName).toMatch(
      /h3/i
    );
    expect(
      getByText('2017-2018 and 2019-2020 election cycles', { exact: false })
    ).not.toBeNull();
  });
});
