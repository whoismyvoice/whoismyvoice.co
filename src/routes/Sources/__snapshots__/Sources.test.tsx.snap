// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="page-block light-white"
>
  <div
    className="faded-bg faded-white hide"
  />
  <div
    className="page-wrapper page-wrapper--sources"
  >
    <h2>
      Sources
    </h2>
    <p
      className="page-introduction"
    >
      The data on this website is based on the data released by the Federal Election Commission (FEC) on donations that were made during the
       
      2017-2018 and 2019-2020
       election cycles. Only donations exceeding $200 in value are required to be publicly disclosed, i.e. (multiple) donations below $200 threshold do not need to be itemized.
    </p>
    <p>
      WhoIsMyVoice.com uses APIs provided by
       
      <a
        href="https://maplight.org/"
      >
        MapLight
      </a>
       to search FEC data and the
       
      <a
        href="https://developers.google.com/civic-information"
      >
        Google Civic Information API
      </a>
       
      to lookup representatives by address. Current legislator information from the Google Civic Information API is matched up with information from
       
      <a
        href="https://theunitedstates.io"
      >
        the @unitedstates project
      </a>
       and raw member data from the
       
      <a
        href="http://clerk.house.gov/xml/lists/MemberData.xml"
      >
        House
      </a>
       
      and
       
      <a
        href="http://www.senate.gov/general/contact_information/senators_cfm.xml"
      >
        Senate
      </a>
      .
    </p>
    <h3>
      For More Information
    </h3>
    <ul
      className="sources"
    >
      <li>
        <a
          href="https://www.cdc.gov/nchs/pressroom/sosmap/firearm_mortality/firearm.htm"
        >
          CDC Statistics on Firearm Mortality By State
        </a>
      </li>
      <li>
        <a
          href="http://www.gunviolencearchive.org/"
        >
          The Gun Violence Archive
        </a>
      </li>
      <li>
        <a
          href="https://everytownresearch.org/"
        >
          Everytown for Gun Safety
        </a>
      </li>
      <li>
        <a
          href="https://home.nra.org/"
        >
          National Rifle Association
        </a>
      </li>
    </ul>
  </div>
</div>
`;
